build/prepare-versioning.ps1

```diff
@@ -0,0 +1,39 @@
+# Updating assemblyInfo and nuspec files with 
+#     build number based on date and time 
+#     and last submission hash from GitHub
+#     and updating copyright information 
+
+$d = Get-Date;
+$date=($d.ToString("yy")-13).ToString() + $d.ToString("MMdd.HHmm");
+$hash = git rev-parse HEAD
+
+$filename = "src\ADAL.Common\CommonAssemblyInfo.cs"
+$content = Get-Content $filename
+$newContent = $content -replace "Microsoft Open Technologies", "Microsoft Corporation"
+$newContent = $newContent + "`n" + "[assembly: AssemblyInformationalVersionAttribute(""$hash"")]"
+
+$m = $newContent -match 'AssemblyFileVersion\(';
+#trying to match something like assembly: AssemblyFileVersion("3.11.0.0").
+$m = $m[0] -match 'AssemblyFileVersion\("([^\.]*.[^\.]*.[^\.]*.[^\"]*)';
+$version = $matches[1]; #API version in the source	
+
+$versionTokens = $version.Split(".");
+$assemblyFileVersion = "{0}.{1}.{2}" -f ($versionTokens[0], $versionTokens[1], $date);
+$newContent = $newContent -replace $version, $assemblyFileVersion
+
+#replace last value with build number. Ex:3.10.0.75
+$assemblyVersion =  "{0}.{1}.{2}.{3}" -f ($versionTokens[0], $versionTokens[1], $versionTokens[2], $env:BUILD_BUILDNUMBER);
```

Comment:
This might indeed cause problems. I encountered issues when setting up the MicroBuild for Dependency Validation. See [here](https://blogs.msdn.microsoft.com/msbuild/2007/01/03/why-are-build-numbers-limited-to-65535/).

VSTS build definitions can be modified to change the format of the build number.